server/channels/jobs/refresh_post_stats/scheduler.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package refresh_post_stats
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/channels/jobs"
+)
+
+func MakeScheduler(jobServer *jobs.JobServer, sqlDriverName string) *jobs.DailyScheduler {
+	startTime := func(cfg *model.Config) *time.Time {
+		parsedTime, err := time.Parse("15:04", *cfg.ServiceSettings.RefreshPostStatsRunTime)
+		if err == nil {
+			return &parsedTime
+		}
+		parsedTime, err = time.Parse("15:04", "00:00")
+		if err != nil {
+			return nil
+		}
+		return &parsedTime
+	}
+	isEnabled := func(cfg *model.Config) bool {
```

Comment:
Do we want any panic button to turn this off? (Feature flag, config value, etc...)

Reply:
We could control it by just having admins unset the refresh time. That would be simplest. Do we think it's necessary? There would also need to be some messaging around that the reporting won't be accurate if you do this.

Reply:
I worry about the "unknown unknowns". Imagine client A that has way more posts/users/whatever than we anticipated, and things start to crash. I would love to have a panic button in that scenario.

Unsetting the refresh time sounds good to me.

Reply:
Okay let's go with that :)

Reply:
How would the admin unset the refresh time?

Reply:
Write a blank string into the admin setting above.